File: Data Structures & Algorithms/daily-temperatures/submission-2.rs
impl Solution {
    pub fn daily_temperatures(temperatures: Vec<i32>) -> Vec<i32> {
        let n = temperatures.len();
        let mut stack = Vec::with_capacity(n);
        let mut more = vec![0; n];

        for (i, &temp) in temperatures.iter().enumerate() {
            while let Some(&last_idx) = stack.last() {
                if temp > unsafe { *temperatures.get_unchecked(last_idx) } {
                    stack.pop();
                    more[last_idx] = (i - last_idx) as i32;
                } else {
                    break;
                }
            }
            stack.push(i);
        }
        
        more
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) == 0:
            return True
        if len(s)%2 != 0:
            return False
        
        map = {"]":"[","}":"{",")":"("}
        if s[0] not in map.values():
            return False
        
        stack = []
        for c in s:
            if c in map:
                if stack and stack[-1] == map[c]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(c)
            
        return stack == []
